server/graphql/programs/src/queries/contact_trace.rs

```diff
@@ -0,0 +1,328 @@
+            ContactTraceSortFieldInput::Gender => ContactTraceSortField::Gender,
+        };
+
+        ContactTraceSort {
+            key,
+            desc: self.desc,
+        }
+    }
+}
+
+#[derive(InputObject, Clone)]
+pub struct ContactTraceFilterInput {
+    pub id: Option<EqualFilterStringInput>,
+    pub program_id: Option<EqualFilterStringInput>,
+    pub document_name: Option<StringFilterInput>,
+    pub datetime: Option<DatetimeFilterInput>,
+    pub patient_id: Option<EqualFilterStringInput>,
+    pub contact_patient_id: Option<EqualFilterStringInput>,
+    pub contact_trace_id: Option<StringFilterInput>,
+    pub first_name: Option<StringFilterInput>,
+    pub last_name: Option<StringFilterInput>,
+    pub gender: Option<EqualFilterGenderInput>,
+    pub date_of_birth: Option<DateFilterInput>,
+}
+
+impl ContactTraceFilterInput {
+    pub fn to_domain_filter(self) -> ContactTraceFilter {
+        let ContactTraceFilterInput {
+            id,
+            program_id,
+            document_name,
+            datetime,
+            patient_id,
+            contact_patient_id,
+            contact_trace_id,
+            first_name,
+            last_name,
+            gender,
+            date_of_birth,
+        } = self;
+        ContactTraceFilter {
+            id: id.map(EqualFilter::from),
+            contact_patient_id: contact_patient_id.map(EqualFilter::from),
+            program_id: program_id.map(EqualFilter::from),
+            document_name: document_name.map(StringFilter::from),
+            datetime: datetime.map(DatetimeFilter::from),
+            contact_trace_id: contact_trace_id.map(StringFilter::from),
+            first_name: first_name.map(StringFilter::from),
+            last_name: last_name.map(StringFilter::from),
+            patient_id: patient_id.map(EqualFilter::from),
+            program_context_id: None,
+            gender: gender.map(|t| map_filter!(t, GenderInput::to_domain)),
+            date_of_birth: date_of_birth.map(DateFilter::from),
+        }
+    }
+}
+
+impl ContactTraceNode {
+    fn trace_row(&self) -> &ContactTraceRow {
+        &self.contact_trace.0
+    }
+
+    fn program_row(&self) -> &ProgramRow {
+        &self.contact_trace.2
+    }
+}
+
+#[Object]
+impl ContactTraceNode {
+    pub async fn id(&self) -> &str {
+        &self.trace_row().id
+    }
+
+    pub async fn store_id(&self) -> Option<String> {
+        self.trace_row().store_id.clone()
+    }
+
+    pub async fn program_id(&self) -> &str {
+        &self.trace_row().program_id
+    }
+
+    pub async fn program(&self) -> ProgramNode {
+        ProgramNode {
+            program_row: self.program_row().clone(),
+        }
+    }
+
+    pub async fn document_id(&self) -> &str {
+        &self.trace_row().document_id
+    }
+
+    pub async fn contact_trace_id(&self) -> Option<String> {
+        self.trace_row().contact_trace_id.clone()
+    }
+
+    pub async fn patient_id(&self) -> &str {
```

Comment:
Yes was wondering about this before. I always came back to leave it in, just in case you just want to have the id and nothing else because then you don't need to do an extra DB query just to get a value you already know...

Reply:
> I always came back to leave it in, just in case you just want to have the id and nothing else because then you don't need to do an extra DB query just to get a value you already know

That was the original reason for having them, but from my understanding this use case never happened, front end logic should only really be about displaying things so it's human readable fields that usually needed, and when `linking` of records by id is requested, it's pretty much always based on users decisions that are made when looking at non id info of the record